Alex/code review.cpp

```diff
@@ -0,0 +1,214 @@
+    
+  }
+  return tempAllPosts;
+  postFile.close();
+}
+
+int main() {
+  vector<User> AllUsers;
+  vector<Post> AllPosts;
+  int UserSelectedOption;
+  string continue_check;
+
+  do {
+    do {
+      cout << "Please select an option: " << endl;
+      cout << "Enter 1: to load data " << "\n" << "Enter 2: to check visibility " << "\n" 
+        << "Enter 3: to retrieve posts " << "\n" << "Enter 4: to search for users by location " 
+        << "\n" << "Enter 5: to terminate program " << "\n";
+      cin >> UserSelectedOption;
+    } while (UserSelectedOption < 1 || UserSelectedOption > 5);
+  
+    
+    if (UserSelectedOption == 1) {
+      string userFilePath_input, postFilePath_input;
+      cout << "Please enter a file path for users: " << "\n";
+      cin >> userFilePath_input;
+      cout << "Please enter a file path for posts: " << "\n";
+      cin >> postFilePath_input;
+
+      AllUsers = loadUserInfo(userFilePath_input);
+      AllPosts = loadPostInfo(postFilePath_input);
+      
+    } else if (UserSelectedOption == 2) {
+      string postID_input, username_input;
+      bool Access = false;
+      cout << "Please enter a post ID: \n";
+      cin >> postID_input;
+      cout << "Please enter a username : \n";
+      cin >> username_input;
+
+      for (Post post: AllPosts) {
+        if (post.postID == postID_input) {
+          if (post.visibility == "public") {
+            Access = true;
+            break;
+          } else if (post.visibility == "friend") {
+            for (User user: AllUsers) {
+              if (user.username == username_input) {
+                if (user.username == post.userID) {
+                  Access = true;
+                  break;
+                } else {
+                  for (const auto & friendName : user.friends) {
+                    if (friendName == post.userID) {
+                      Access = true;
+                      break;
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+
+      if (Access == true) {
+          cout << "Access Permitted \n";
+      } else if (Access == false) {
+          cout << "Access Denied \n";
+      }
+      
+    } else if (UserSelectedOption == 3) {
+      string userName_input;
+      cout << "Please enter a user name: \n";
+      cin >> userName_input;
+
+      for (Post post: AllPosts) {
+        if (post.userID == userName_input){
+          cout << post.postID << "\n";
+        }
+      }
+  
+    } else if (UserSelectedOption == 4) {
+      string location_input;
+      cout << "Please enter a location: \n";
+      cin >> location_input;
+
+      for (User user: AllUsers) {
+        if (user.state == location_input) {
+          cout << user.displayname << "\n";
+        }
+      }
+      
+    } else if (UserSelectedOption == 5) {
+      cout << "Program Terminated..." << endl;
+      break;
+    }
+
+    cout << "Would you like to continue? (y/n) \n";
```

Comment:
Simple yes or no to continue the loop is good, with your error statements being well placed. Either from an exact error statement or just not showing a response based on the false information added.